plugins/woocommerce/src/Internal/Features/FeaturesController.php

```diff
@@ -722,16 +703,23 @@ private function get_setting_for_feature( string $feature_id, array $feature, bo
 		 */
 		$desc_tip = apply_filters( 'woocommerce_feature_description_tip', $desc_tip, $feature_id, $disabled );
 
-		return array(
+		$feature_setting = array(
 			'title'    => $feature['name'],
 			'desc'     => $description,
-			'type'     => 'checkbox',
+			'type'     => $type,
 			'id'       => $this->feature_enable_option_name( $feature_id ),
 			'disabled' => $disabled && ! $this->force_allow_enabling_features,
 			'desc_tip' => $desc_tip,
 			'tooltip'  => $tooltip,
 			'default'  => $this->feature_is_enabled_by_default( $feature_id ) ? 'yes' : 'no',
 		);
+
+		/**
+		 * Allows to modify feature setting that will be used to render in the feature page.
+		 *
+		 * @since 8.0.0
```

Comment:
Should we document the filter params?

```suggestion
		 * @since 8.0.0
		 *
		 * @param array $feature_setting {
		 *     Describes the feature.
		 *
		 *     @type string $title    Feature title.
		 *     @type string $desc     Feature description.
		 *     @type string $type     Example: 'checkbox'.
		 *     @type string $id       Feature ID/slug.
		 *     @type bool   $disabled If disabled.
		 *     @type string $desc_tip Description tooltip text.
		 *     @type string $default  Default value for this feature setting.
		 * }
		 * @param string $feature_id  The feature ID.
		 
```

Reply:
Addressed in https://github.com/woocommerce/woocommerce/pull/38993/commits/94fce9c169809e484a9d75803600ae3ad88b4ee5